### ISSUE TITLE
clarify dynamic composition via compose

### ISSUE THREAD

**Issue by simonfox** [aurelia/documentation#57]
Coming from Durandal, dynamic composition allowed a view model instance to be passed directly to the compose binding and the view would be determined dynamically based on the type of view model.

It seems this approach is not supported in Aurelia, instead you bind to a model and specify the type of view model that the compose Custom Element should create to wrap (as it is passed to activate) that model.

Is my understanding correct?
If it is, what were the reasons for not including this ability in Aurelia?

Assuming my understanding is correct, it might be worth adding a note to the docs for Durandal users to explicitly state this difference. I'd be happy to contribute this, but would like to know the reasons its not included first.

**Comment by EisenbergEffect** [comment]
So, we intend to have this in Aurelia. The problem lies mostly with the way that the ES6 Module Loader spec is written. With Durandal, we were based on require.js and were able to plug in and do a little magic that enabled these dynamic scenarios. With the ES6 loader....things are locked up a bit tighter. However, there are some additions to the spec which would enable use to make this work in the future. I'm not sure when those will land, but when they do, we will fix this up.

So, in short, it's going to work, but at the moment it doesn't due to a spec-related issue. We have another issue that is tracking this, but I'll leave this open too since it's a bit clearer here.

**Comment by simonfox** [comment]
Awesome news, thanks @EisenbergEffect 

I'll have a look for the other issue and track that as well...

**Comment by simonfox** [comment]
For anyone interested, the related issue tracking this is https://github.com/aurelia/templating-resources/issues/17

**Comment by EisenbergEffect** [comment]
@simonfox I'm going to close this issue and we'll just track this through the one referenced above.
